From foo@baz Mon Apr 18 10:31:57 JST 2016
Date: Mon, 18 Apr 2016 10:31:57 +0900
To: Greg KH <gregkh@linuxfoundation.org>
From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Subject: Revert bad backport of "drm/radeon: hold reference to fences in radeon_sa_bo_new"

From: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

This reverts commit 50353e6f86eb2ac46ffe3cc0b9f9a11ddc8a9410, which is
commit f6ff4f67cdf8455d0a4226eeeaf5af17c37d05eb upstream, as it was
backported to the 3.14-stable tree incorrectly.  A correct fix will
happen next.

Reported-by: Nicolai Hähnle <nicolai.haehnle@amd.com>
Cc: Christian König <christian.koenig@amd.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>


---
 drivers/gpu/drm/radeon/radeon_sa.c |    5 -----
 1 file changed, 5 deletions(-)

--- a/drivers/gpu/drm/radeon/radeon_sa.c
+++ b/drivers/gpu/drm/radeon/radeon_sa.c
@@ -349,13 +349,8 @@ int radeon_sa_bo_new(struct radeon_devic
 			/* see if we can skip over some allocations */
 		} while (radeon_sa_bo_next_hole(sa_manager, fences, tries));
 
-		for (i = 0; i < RADEON_NUM_RINGS; ++i)
-			radeon_fence_ref(fences[i]);
-
 		spin_unlock(&sa_manager->wq.lock);
 		r = radeon_fence_wait_any(rdev, fences, false);
-		for (i = 0; i < RADEON_NUM_RINGS; ++i)
-			radeon_fence_unref(&fences[i]);
 		spin_lock(&sa_manager->wq.lock);
 		/* if we have nothing to wait for block */
 		if (r == -ENOENT && block) {
